pymeasure/instruments/keysight/keysight34980A.py

```diff
@@ -0,0 +1,628 @@
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:CURRent:AC?")
+        return(str(temp_values[0]))
+
+    def measure_frequency(self, chan, resolution = "DEFault", range_param = "DEFault"):    
+        """
+        The following command configures the instrument for frequency measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:FREQuency?")
+        return(str(temp_values[0]))
+
+    def measure_period(self, chan, resolution = "DEFault", range_param = "DEFault"):    
+        """
+        The following command configures the instrument for period measurements, 
+        triggers the internal DMM to scan the specified channel.    
+        """
+        temp_values = self.values(":MEASure:SCALar:PERiod?")
+        return(str(temp_values[0]))
+
+    def measure_resistance(self, chan, resolution = "DEFault", range_param = "AUTO"):    
+        """
+        The following command configures the instrument for resistance measurements, 
+        triggers the internal DMM to scan the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:RESistance?")
+        return(str(temp_values[0]))
+
+    def measure_temperature(self, chan, probe = "TCouple", probe_type = "T", resolution = "DEFault", range_param = "AUTO"):   
+        """
+        The following command configures the instrument for temperature measurements on the specified channel.
+        """
+        temp_values = self.values(":MEASure:SCALar:TEMPerature?")
+        return(str(temp_values[0]))
+
+    def set_temperature_unit(self, chan = "", unit = 'K'):    
+        """
+        This command selects the temperature units (°C, °F, or Kelvins) on the specified channels. 
+        If you omit the optional <ch_list> parameter, this command applies to the internal DMM, 
+        independent of any channels or a scan list.
+        """
+        if (chan == ""):
+            self.write(":UNIT:TEMPerature " + str(unit))
+            while((self.is_temperature_unit()) != str(str(unit) + "\n")):
+                self.write(":UNIT:TEMPerature " + str(unit))    
+        else:
+            self.write(":UNIT:TEMPerature " + str(unit) + ", (@" + str(chan) + ")")
+            while((self.is_temperature_unit(chan)) != str(str(unit) + "\n")):
+                self.write(":UNIT:TEMPerature " + str(unit) + ", (@" + str(chan) + ")")
+
+    def is_temperature_unit(self, chan = ""):    
+        """
+        The command checks the temperature units (°C, °F, or Kelvins) on the specified channels. 
+        If you omit the optional <ch_list> parameter, this command applies to the internal DMM, 
+        independent of any channels or a scan list. 
+        """
+        if (chan == ""):
+            self.write(":UNIT:TEMPerature?") 
+            temp_values = self.read()
+            return(str(temp_values))
```

Comment:
Structures like this (i.e. writing a certain query, reading the response and casting it into a certain type) can be best performed using a `self.ask` (in case you expect a single string to be returned) or `self.values` (in case you expect one or more numeric values to be returned).